src/fsharp/IlxGen.fs

```diff
@@ -5892,10 +6136,13 @@ and GenMethodForBinding
 
         | _ ->
             // Replace the body of ValInline.PseudoVal "must inline" methods with a 'throw'
-            // However still generate the code for reflection etc.
+            // For witness-passing methods, don't do this if `isLegacy` flag specified
+            // on the attribute. Older compilers
             let bodyExpr =
-                if HasFSharpAttribute g g.attrib_NoDynamicInvocationAttribute v.Attribs then
-                    let exnArg = mkString g m (FSComp.SR.ilDynamicInvocationNotSupported(v.CompiledName g.CompilerGlobalState))
+                let attr = TryFindFSharpBoolAttributeAssumeFalse cenv.g cenv.g.attrib_NoDynamicInvocationAttribute v.Attribs
+                if (not generateWitnessArgs && attr.IsSome) ||
```

Comment:
nit: Pattern match be better?
```fsharp
match attr, generateWitnessArgs with
| Some _, false
| Some false, true ->
    ...
| _ ->
    ...
```

Reply:
In this case I prefer the if/then/else I think